### DISCUSSION TITLE
Is there a list anywhere of the needed game files?

### DISCUSSION THREAD

**Discussion by Kirtai** [ihhub/fheroes2#3486] [Q&A]
Is there a list of which files are required from the original game? I have the GOG version if it matters. A list would be handy to make packages from.

**Comment by ihhub** [comment]
Hi @Kirtai , which OS are you referring to in terms of packages?

**Reply by Kirtai** [reply]
It's for Linux. 

There are projects which allow people to easily build installable packages from their own games to install on their system.

Two I know of are [game-data-packager](https://salsa.debian.org/games-team/game-data-packager) and [play.it](https://forge.dotslashplay.it/play.it/scripts).

It would be nice to add HoMM2 & fheroes to one of them.

To compare, see how HoMM3 is already supported in g-d-p with the vcmi engine and play.it via dosbox

**Reply by ihhub** [reply]
The only company which has rights to do packaging with file resources/assets is Ubisoft. We don't have such rights to distribute packages with original files. The only way is to download demo version of the game as a part of provided script or an application but not to include it into package. Alternatively, which should be verified is to run a separate script on Linux machine which contains HoMM2 original files, copy them to a separate place and unzip fheroes2 there.

That's why download_demo.sh script is included into our archives.

**Reply by Kirtai** [reply]
Sorry, I see I wasn't clear. These are for _owners of the game_ to convert the game into a package that the system package manager can handle instead of just putting loose files somewhere.

For example, I ran "game-data-packager heroes3" and it downloaded my purchased game from GOG (via lgogdownloader), extracted it and build "homm3-en-data_67_all.deb" which I can then install system wide with "sudo apt install homm3-en-data_67_all.deb".

Also, I find it very untidy to leave unnecessary files like old EXEs and redistributables littering the game data directory so would like to know what can be removed :)

**Reply by ihhub** [reply]
The described case can lead to distribution of debian packages online. Also we have a [pull request](https://github.com/ihhub/fheroes2/pull/2953) for Debian package creation without resource files. Maybe we could modify it to download resource files from GOG after package installation.

**Reply by Kirtai** [reply]
All I want is to know which files I can keep and which I can delete from my HoMM2 install without it breaking with fheroes.

The packaging thing is strictly a convenience for game owners and was just an example of why I want it. (Those tools only work with games you _already have_).

BTW, gdp is from Debian's official games team. It's not a piracy tool.

**Reply by ihhub** [reply]
In short fheroes2 uses all files stored in directories such as data, music, maps, heroes2.

**Reply by Kirtai** [reply]
Thank you :)

(The GOG version is messy)

**Reply by ihhub** [reply]
You're welcome! Also if you have some comments about the existing Debian packaging pull request please feel free to leave comments.

**Reply by qarkai** [reply]
> In short fheroes2 uses all files stored in directories such as data, music, maps, heroes2.

It would be nice if this list was in README somewhere.
